src/patternfly/components/FileUpload/examples/FileUpload.md

```diff
@@ -0,0 +1,150 @@
+---
+title: File upload
+section: beta
+cssPrefix: pf-c-file-upload
+---
+
+## Examples
+
+```hbs title=Basic-file-upload
+{{#> file-upload file-upload--id="basic-file-upload"}}
+  {{#> file-upload-input-group}}
+    {{#> input-group}}
+      {{> file-upload-text-input
```

Comment:
This was discussed with design, the user has the ability to change the name of the file. But im sure that with the React implementation we will be able to analyze if this is something that we want.

Reply:
Good question @matthewcarleton . I may be misremembering, but didn't we say that the user should not be able to enter a file name directly into this field as we wanted the browser to handle any access to the user's file system @christiemolloy . Can you confirm @mturley ? If that's the case, maybe this should be made read-only at all times.

Reply:
ya I agree with you @mcarrano. It seems unlikely that they'd need to edit the name if they put the wrong file in there. I would guess (maybe there's research out there to back this up) most users don't expect to be able to edit a filename for this type of interaction.

Reply:
Yes @mcarrano you're correct, that's what we discussed. In the implementation, it won't make sense to allow the user to edit a filename here, because it's not clear what effect that would have.

If the user either presses the Browse button and selects a file, or drags and drops a file into the component, then the read-only filename is populated from the selected file. If it were possible to change, it would have no effect, the file was already loaded.

I can only think of two things typing a filename in that box would do in theory: (a) the user manually specifies a path to a file on disk instead of browsing for it, or (b) the user is entering the name of a file to be created somewhere in the consuming app as a result of this form submission. (a) is not possible due to security restrictions (we need the browser to provide the file path and contents based on user action, JavaScript can't just ask the browser for the contents of an arbitrary file path), and (b) did not seem to be a need here.

In the future if we want to support a use case like (b) we could add a prop to make that field editable as a non-breaking change.